src/main/java/org/sunbird/progress/service/MandatoryContentServiceImpl.java

```diff
@@ -225,39 +245,54 @@ private UserProgressRequest validateGetBatchEnrolment(SunbirdApiRequest requestB
+		if (!ObjectUtils.isEmpty(userObj)) {
+			responseObj.put(Constants.FIRSTNAME, userObj.getFirstName());
+			responseObj.put(Constants.LASTNAME, userObj.getLastName());
+			responseObj.put(Constants.DEPARTMENT, userObj.getChannel());
 
-				if (!ObjectUtils.isEmpty(userObj.getProfileDetails())) {
-					SunbirdUserProfileDetail profileDetails = userObj.getProfileDetails();
-					if (!ObjectUtils.isEmpty(profileDetails.getPersonalDetails())
-							&& profileDetails.getPersonalDetails().containsKey(Constants.PRIMARY_EMAIL)) {
-						responseObj.put(Constants.EMAIL,
-								profileDetails.getPersonalDetails().get(Constants.PRIMARY_EMAIL));
-					}
-					if (!ObjectUtils.isEmpty(profileDetails.getProfessionalDetails())
-							&& !profileDetails.getProfessionalDetails().isEmpty()) {
-						if (!ObjectUtils.isEmpty(profileDetails.getProfessionalDetails().get(0).getDesignation())) {
-							responseObj.put(Constants.DESIGNATION,
-									profileDetails.getProfessionalDetails().get(0).getDesignation());
-						} else if (!ObjectUtils
-								.isEmpty(profileDetails.getProfessionalDetails().get(0).getDesignationOther())) {
-							responseObj.put(Constants.DESIGNATION,
-									profileDetails.getProfessionalDetails().get(0).getDesignationOther());
-						} else {
-							responseObj.put(Constants.DESIGNATION, "");
-						}
+			if (!ObjectUtils.isEmpty(userObj.getProfileDetails())) {
+				SunbirdUserProfileDetail profileDetails = userObj.getProfileDetails();
+				if (!ObjectUtils.isEmpty(profileDetails.getPersonalDetails())
+						&& profileDetails.getPersonalDetails().containsKey(Constants.PRIMARY_EMAIL)) {
+					responseObj.put(Constants.EMAIL, profileDetails.getPersonalDetails().get(Constants.PRIMARY_EMAIL));
+				}
+				if (!ObjectUtils.isEmpty(profileDetails.getProfessionalDetails())
+						&& !profileDetails.getProfessionalDetails().isEmpty()) {
+					if (!ObjectUtils.isEmpty(profileDetails.getProfessionalDetails().get(0).getDesignation())) {
+						responseObj.put(Constants.DESIGNATION,
+								profileDetails.getProfessionalDetails().get(0).getDesignation());
+					} else if (!ObjectUtils
+							.isEmpty(profileDetails.getProfessionalDetails().get(0).getDesignationOther())) {
+						responseObj.put(Constants.DESIGNATION,
+								profileDetails.getProfessionalDetails().get(0).getDesignationOther());
 					} else {
 						responseObj.put(Constants.DESIGNATION, "");
 					}
+				} else {
+					responseObj.put(Constants.DESIGNATION, "");
 				}
 			}
 		}
 	}
+
+	private int getCompletionPercentage(int progress, int leafNodeCount) {
+		if (progress == 0) {
+			return 0;
+		} else if (progress >= 1 && progress < leafNodeCount) {
+			return (progress * 100) / leafNodeCount;
+		} else {
+			return 100;
+		}
+	}
+
+	private int getCompletionStatus(int progress, int leafNodeCount) {
+		if (progress == 0) {
+			return 0;
+		} else if (progress >= 1 && progress < leafNodeCount) {
+			return 1;
+		} else {
+			return 2;
+		}
+	}
```

Comment:
this code is duplicated. Just have one method which accepts the responseObj too... 
get the progress from responseObj and compare progress & leafNodeCount -- update the progress and status directly to responseObj